suite-common/wallet-core/src/discovery/discoveryThunks.ts

```diff
@@ -279,58 +265,64 @@ export const getBundleThunk = createThunk(
+                        account.accountType === accountType && account.symbol === networkSymbol,
                 )
                 .sort((a, b) => b.index - a.index);
 
             // check if requested coin already have an empty account
             const hasEmptyAccount = prevAccounts.find(a => a.empty && !a.visible);
             // check if requested coin not failed before
             const failed = discovery.failed.find(
-                account =>
-                    account.symbol === configNetwork.symbol && account.accountType === accountType,
+                account => account.symbol === networkSymbol && account.accountType === accountType,
             );
 
             // skip legacy/ledger accounts if availableCardanoDerivations doesn't include their respective derivation
             const skipCardanoDerivation =
-                configNetwork.networkType === 'cardano' &&
-                (configNetwork.accountType === 'ledger' ||
-                    configNetwork.accountType === 'legacy') &&
-                !discovery.availableCardanoDerivations?.includes(configNetwork.accountType);
+                networkType === 'cardano' &&
+                (accountType === 'ledger' || accountType === 'legacy') &&
+                !discovery.availableCardanoDerivations?.includes(accountType);
 
             if (!hasEmptyAccount && !failed && !skipCardanoDerivation) {
                 const index = prevAccounts[0] ? prevAccounts[0].index + 1 : 0;
                 const isEvmLedgerDerivationPath =
-                    configNetwork.networkType === 'ethereum' &&
-                    configNetwork.accountType === 'ledger';
+                    networkType === 'ethereum' && accountType === 'ledger';
 
                 const pathIndex = (index + (isEvmLedgerDerivationPath ? 1 : 0)).toString();
 
                 bundle.push({
-                    path: configNetwork.bip43Path.replace('i', pathIndex),
-                    coin: configNetwork.symbol,
+                    path: bip43Path.replace('i', pathIndex),
+                    coin: networkSymbol,
                     identity: tryGetAccountIdentity({
-                        networkType: configNetwork.networkType,
+                        networkType,
                         deviceState: discovery.deviceState,
                     }),
                     details: 'txs',
                     index,
-                    pageSize: getTxsPerPage(configNetwork.networkType),
+                    pageSize: getTxsPerPage(networkType),
                     accountType,
-                    networkType: configNetwork.networkType,
+                    networkType,
                     derivationType: getDerivationType(accountType),
                     suppressBackupWarning: true,
                 });
             }
-        });
+        };
+
+        // foreach network, start discovery foreach normalized account (normal as well as all alternative accounts)
```

Comment:
"normalized account" is a very helpful helper introduced in [#14281 here](https://github.com/trezor/trezor-suite/commit/8f0270267d714f747542facdee446861ef20ed69), and used in `filterUnavailableAccountTypes`.

TL;DR: It's kinda like `networksCompatibility` but **only** for accounts, which actually does make sense! It gets accounts of network, but synthesizes a `'normal'` account from the network properties, and for the alternative accs it backfills properties.
:arrow_right: never again I have to do stuff like `account.bip43Path ?? network.bip43Path`, because it is very easy to do a mistake there. You just iterate over all accounts, and the correct properties are guaranteed to be there :ok_hand:

Reply:
Thank you! and for tests too!